### PR TITLE
Subject property issue

### PR DESCRIPTION
**Pull request by njaffer**

### CODE REVIEW

File: app/forms/curation_concerns/forms/work_form.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
+module CurationConcerns
+  module Forms
+    class WorkForm
```
Comment (grosscol) [inline app/forms/curation_concerns/forms/work_form.rb]: I don't understand why this class is being added.  What is this form used for?

File: app/views/generic_files/_subject_modal.html.erb [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<div class="modal-div">
+<!-- Modal -->
```
Comment (grosscol) [inline app/views/generic_files/_subject_modal.html.erb]: Where is this modal invoked from?

File: app/views/generic_files/_subject_modal.html.erb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+<div class="modal fade" id="rightsModal" tabindex="-1" role="dialog" aria-labelledby="myModalLabel" aria-hidden="true">
+  <div class="modal-dialog">
+    <div class="modal-content">
+      <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal" aria-hidden="true"></button>
+        <h2 id="rightsModallLabel"><%= t('sufia.product_name') %> License Descriptions</h2>
+      </div>
+      <div class="modal-body">
```
Comment (grosscol) [inline app/views/generic_files/_subject_modal.html.erb]: Should the contents of the modal be the rights agreements?

File: app/forms/curation_concerns/forms/work_form.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+          else
+            super
+          end
+        end
+
+        # Overriden to cast 'rights' to an array
+        def sanitize_params(form_params)
+          super.tap do |params|
+            params['rights'] = Array(params['rights']) if params.key?('rights')
+            params['subject'] = Array(params['subject']) if params.key?('subject')
```
Comment (grosscol) [inline app/forms/curation_concerns/forms/work_form.rb]: Is this entire file being included in order to alter this line?

Comment (njaffer) [inline app/forms/curation_concerns/forms/work_form.rb]: Yes!

Comment (grosscol) [inline app/forms/curation_concerns/forms/work_form.rb]: I think there are more concise methods for modifying a single class method.   Base ruby has a few idioms to choose from, but rails might have a convenient pattern for it as well.

**Comment by grosscol** [comment]
Perhaps approaching this as the Rights are is more complicated that warranted for the initial deployment.  A more simplified approach could be done in the same way that the resource type functions.  We can refactor to use Qa when we have uri representations of the high level browse subjects.

File: app/forms/curation_concerns/forms/work_form.rb (hunk 4 above)
Comment (njaffer) [inline app/forms/curation_concerns/forms/work_form.rb]: yeah I am looking at that..

File: app/views/generic_files/_subject_modal.html.erb (hunk 2 above)
Comment (njaffer) [inline app/views/generic_files/_subject_modal.html.erb]: I noticed that both the erb files are missing from this commit.. adding those now

File: app/views/generic_files/_subject_modal.html.erb (hunk 3 above)
Comment (njaffer) [inline app/views/generic_files/_subject_modal.html.erb]: editing this file too. This is called when the user clicks on help text button

**Pull request merged**